### ISSUE TITLE
Training on new dataset

### ISSUE THREAD

**Issue by tanzheen** [bytedance/1d-tokenizer#36]
Hi I am looking to train this tokenizer on a sign language dataset CSL-Daily so that I can compress video frames into 32 tokens. 
However, I am not getting very good results (attached below). I would like to check if all I have to do is download the pretrained model and then run it as: 
```
WANDB_MODE=offline accelerate launch --num_machines=1 --num_processes=8 --machine_rank=0 --main_process_ip=127.0.0.1 --main_process_port=9999 --same_network scripts/train_titok.py config=configs/training/stage1/titok_b64.yaml \
    experiment.project="titok_b64_stage1" \
    experiment.name="titok_b64_stage1_run1" \
    experiment.output_dir="titok_b64_stage1_run1" \
    training.per_gpu_batch_size=32
``` 
![00862136_s-001](https://github.com/user-attachments/assets/72b89c15-9a9b-42c1-aa06-d801054ad363)
![00862136_s-000](https://github.com/user-attachments/assets/c0ff544a-3159-4f48-ba74-3d72c0a1691a)

Please advice! Thanks!

**Comment by cornettoyu** [comment]
> Hi I am looking to train this tokenizer on a sign language dataset CSL-Daily so that I can compress video frames into 32 tokens. However, I am not getting very good results (attached below). I would like to check if all I have to do is download the pretrained model and then run it as:
> 
> ```
> WANDB_MODE=offline accelerate launch --num_machines=1 --num_processes=2 --machine_rank=0 --main_process_ip=127.0.0.1 --main_process_port=9999 --same_network scripts/train_titok.py config=configs/training/stage1/titok_;32.yaml \
>     experiment.project="titok_l32_stage1" \
>     experiment.name="titok_;32_stage1_run1" \
>     experiment.output_dir="titok_l32_stage1_run1" \
>     training.per_gpu_batch_size=32
> ```
> 
> ![00862136_s-001](https://private-user-images.githubusercontent.com/85047594/375217055-72b89c15-9a9b-42c1-aa06-d801054ad363.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTgyOTMsIm5iZiI6MTcyODU5Nzk5MywicGF0aCI6Ii84NTA0NzU5NC8zNzUyMTcwNTUtNzJiODljMTUtOWE5Yi00MmMxLWFhMDYtZDgwMTA1NGFkMzYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDIyMDYzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4YmY1OTZjYTY4OWYwYTAyYzYwOGZkNDdlZWFjOWU3ODI3MjY1ZjQ2MTBjNmVmNzcwOTZmOWVmMjk5YjZmMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8iwrmmcjUnvV9q2S_Qww3h-iaBr3J4wIbUKV-sujd7s) ![00862136_s-000](https://private-user-images.githubusercontent.com/85047594/375217106-c0ff544a-3159-4f48-ba74-3d72c0a1691a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1OTgyOTMsIm5iZiI6MTcyODU5Nzk5MywicGF0aCI6Ii84NTA0NzU5NC8zNzUyMTcxMDYtYzBmZjU0NGEtMzE1OS00ZjQ4LWJhNzQtM2Q3MmMwYTE2OTFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDIyMDYzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliNTczYzYzNDIyNjQzYmI5ZGVkODk0YzM5Y2RiMDJjODMwZGY1ZjBmMmYzNmE2ODYwYTRiNjliMzY4YTFjYjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4klgU-7GsjRIyqdOCoaNQtsROw6X_7PHQ5lCF_sc4j0)
> 
> Please advice! Thanks!

It seems to me that you only did stage1 training, where I would expect some blurry faces as it relies on the ImageNet-trained MaskGIT-VQGAN which is not good for humans. Did you try run stage2 fine-tuning as well? I would expect the stage2 finetuning should help in your case

**Comment by tanzheen** [comment]
Yes, I only did stage 1 training, this is due to the fact that I only want the new 32 embedded tokens. Will the stage 2 training affect the values of the 32 tokens? I read in the paper that stage 2 training is only for the decoder

Also, do you think I should train MaskGIt to my dataset too for the proxy codes?

**Comment by tanzheen** [comment]
@cornettoyu do you have any inputs

**Comment by cornettoyu** [comment]
> @cornettoyu do you have any inputs

IMHO it would still be helpful to train stage2, to see if the reconsturction results can be better. If not, then it means the token representation is bottlenecked by gap between your dataset and ImageNet, then I would suggest a full fine-tuning on you dataset instead of decoder only (e.g., you can enable the gradient for encoder/quantizer [here](https://github.com/bytedance/1d-tokenizer/blob/main/modeling/titok.py#L100))

**Comment by grsilva9** [comment]
Hey,

First of all, congratulations on the great work. Absolutely outstanding.

On the topic of training on a new dataset from scratch, what are the steps/command I should use to train stages 1 and 2?

I'd really appreciate some directions on this.
